core/consensus/babe/impl/babe_impl.cpp

```diff
@@ -158,11 +158,45 @@ namespace kagome::consensus::babe {
     return current_state_;
   }
 
+  void BabeImpl::onRemoteStatus(const libp2p::peer::PeerId &peer_id,
+                                const network::Status &status) {
+    if (current_state_ == State::WAIT_BLOCK) {
+      if (status.best_block.number == 1) {
+        current_state_ = State::SYNCHRONIZED;
+        return;
+      }
+
+      const auto &last_finalized_block = block_tree_->getLastFinalized();
+
+      auto current_best_block_res = block_tree_->getBestContaining(
+          last_finalized_block.hash, boost::none);
+      BOOST_ASSERT(current_best_block_res.has_value());
+      const auto &current_best_block = current_best_block_res.value();
```

Comment:
I suggest adding an if statement alongside to assert to log an error when kagome is built in release mode and then return from method gracefully.

Reply:
Best block always is descendant of finalized block by definition.